### ISSUE TITLE
Categories applied to the ElementDefinition ignored when calculating budget of ElementUsages

### ISSUE THREAD

**Issue by alexatrhea** [RHEAGROUP/CDP4-IME-Community-Edition#299]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When computing budget, if the appropriate category has been applied to an ED, it is not used for lookup. It has to be reapplied to the EU separately.

EU categories in principle should be a union of own applied categories and those of the ED it derives from.

### Steps to Reproduce
Set up a normal budget scenario but only apply the relevant categories to the various element definitions. Notice that the budget cannot compute. Apply same categories to the EUs directly and see that the budget can now be computed.

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- CDP4 version:
  - [ ] CDP4-IME: 
  - [ ] Other: 
  -
- Web Services:
  - [ ] CDP4 Web Services version: 
  - [ ] OCDT WSP version: 
- Environment (Operating system, version and so on):
  - [ ] Microsoft Excel (32/64 bit): 
  - [ ] Windows OS (32/64 bit): 
  - [ ] Additional information:

**Comment by samatrhea** [comment]
sounds like the current implementation is not using the convenience method that makes a union of ED and UD applied categories: CDP4-SDK (Common) `CategorizableThingExtensions.GetAllCategories`

**Comment by alexatrhea** [comment]
@samatrhea Indeed

**Comment by samatrhea** [comment]
@lxatrhea  and @alexatrhea, is this still an issue?

**Comment by lxatrhea** [comment]
No. Looks like line 331 in BudgetConfigViewModel only takes the ElementBase's categories:

```
this.usedCategories = new List<Category>(elementBases.SelectMany(x => x.Category).Distinct().OrderBy(x => x.Name));
```
